src/docs/tool-libfuzzer.dox

```diff
@@ -0,0 +1,210 @@
+@endcode
+
+## Step 2: Build as usual
+
+@code
+$ cd build_posix
+$ make
+@endcode
+
+## Step 3: Run a fuzz target
+
+Each fuzz target is just a program under \c build_posix/test/fuzz/. Once you become familiar with
+LibFuzzer, it may be easier to run the target program directly, but for first-time users, the script
+at \c test/fuzz/fuzz_run.sh is handy. It performs a limited number of runs, automatically cleans up
+after itself in between runs and provides a sensible set of parameters which you can add to. For
+example:
+
+@code
+$ cd build_posix/test/fuzz/
+$ bash ../../../test/fuzz/fuzz_run.sh ./fuzz_modify
+@endcode
+
+In general the usage is:  
+
+@code
+fuzz_run.sh <fuzz-test-binary> [fuzz-test-args]
+@endcode
+
+Each fuzz target will produce a few outputs:
+
+- \c fuzz-N.log: The LibFuzzer log for worker N.
+
+- \c WT_TEST_<pid>: The home directory for a given worker process.
+
+- \c crash-<input-hash>: If an error occurs, you may see a file like this containing the input that
+crashed the target.
+
+- \c WT_TEST_<pid>.profraw: If you're running with Clang coverage (more on this later), you'll see
+files containing profiling data for a given worker. These will be used by \c fuzz_coverage.
+
+### Corpus
+
+LibFuzzer is a coverage based fuzzer meaning that it notices when a particular input hits a new code
+path and adds it to a "corpus" of interesting data inputs. It then uses existing data in the corpus
+to mutate and come up with new inputs.
+
+While LibFuzzer will automatically add to its corpus when it finds an interesting input, some fuzz
+targets (especially those that expect data in a certain format) require a corpus to start things off
+in order to be effective. The target \c fuzz_config is one example of this as it expects its data
+sliced with a separator so the fuzzer needs some examples to guide it. The corpus is supplied as the
+first positional argument to both \c fuzz_run.sh and the underlying fuzz target itself. For example:
+
+@code
+$ cd build_posix/test/fuzz/
+$ bash ../../../test/fuzz/fuzz_run.sh ./fuzz_config ../../../test/fuzz/config/corpus/
+@endcode
+
+# Implementing an LLVM LibFuzzer target
+
+## Overview
+
+So you have functionality that you'd like to test against LLVM LibFuzzer and would like to add a new
+target to the system.
+
+We do this by implementing a function with the following prototype:
+
+@code
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+@endcode
+
+When we supply the \c -fsanitize=fuzzer flag, Clang will use its own \c main function and repeatedly
+call your \c LLVMFuzzerTestOneInput with various inputs. There is a lot of information online about
+best practices when writing fuzzing targets but to summarize, the requirements are much like those
+for writing unit tests: they should be fast, deterministic and stateless (as possible). The
+[LLVM LibFuzzer reference page](https://llvm.org/docs/LibFuzzer.html) is a good place to start if
+you'd like to learn more.
```

Comment:
I suggest: "a good place to start to learn more."

Reply:
7a17f05